src/org/labkey/test/util/DataClassHelper.java

```diff
@@ -0,0 +1,286 @@
+        clickAndWait(Locator.linkWithText(name));
+        return this;
+    }
+
+    public void verifyFields(List<FieldDefinition> _fields)
+    {
+        TestLogger.log("Verify that the fields for the data class are as expected");
+        Set<String> actualNames = new HashSet<>(getDataClassFields());
+        Set<String> expectedNames = _fields.stream().map(FieldDefinition::getName).collect(Collectors.toSet());
+        expectedNames.add("Name");
+        expectedNames.add("Description");
+        assertEquals("Fields in data class not as expected.", expectedNames, actualNames);
+    }
+
+    public DataRegionTable getDataClassesList()
+    {
+        return new DataRegionTable.DataRegionFinder(getDriver()).withName(DATA_CLASS_DATA_REGION_NAME).find();
+    }
+
+    public DataRegionTable getDataClassDataRegionTable()
+    {
+        return new DataRegionTable.DataRegionFinder(getDriver()).withName("query").find();
+    }
+
+    public int getDataCount()
+    {
+        return getDataClassDataRegionTable().getDataRowCount();
+    }
+
+    private List<String> getDataClassFields()
+    {
+        return getDataClassDataRegionTable().getColumnNames();
+    }
+
+    public void insertRow(Map<String, String> fieldValues)
+    {
+        getDataClassDataRegionTable()
+                .clickInsertNewRow();
+        for (Map.Entry<String, String> fieldValue : fieldValues.entrySet())
+        {
+            setFormElement(Locator.name("quf_" + fieldValue.getKey()), fieldValue.getValue());
+        }
+        clickButton("Submit");
+    }
+
+    public void bulkImport(File dataFile)
+    {
+        fileImport(dataFile, IMPORT_OPTION);
+    }
+
+    public void mergeImport(File dataFile)
+    {
+        fileImport(dataFile, MERGE_OPTION);
+    }
+
+    private void fileImport(File dataFile, String importOption)
+    {
+        DataRegionTable drt = getDataClassDataRegionTable();
+        TestLogger.log("Adding data from file");
+        ImportDataPage importDataPage = drt.clickImportBulkData();
+        importDataPage.setFile(dataFile);
+        if (MERGE_OPTION.equals(importOption))
+            importDataPage.setFileMerge(true);
+        else if (UPDATE_OPTION.equals(importOption))
+            importDataPage.setFileInsertOption(true);
+        importDataPage.submit();
+    }
+
+    public void bulkImport(String tsvData)
+    {
+        startTsvImport(tsvData, IMPORT_OPTION)
+                .submit();
+    }
+
+    public void mergeImport(String tsvData)
+    {
+        startTsvImport(tsvData, DataClassHelper.MERGE_OPTION)
+                .submit();
+    }
+
+    public void bulkImport(List<Map<String, String>> data)
+    {
+        startTsvImport(data, IMPORT_OPTION)
+                .submit();
+    }
+
+    public void bulkImportExpectingError(List<Map<String, String>> data, String importOption)
+    {
+        startTsvImport(data, importOption)
+                .submitExpectingError();
+    }
```

Comment:
I know the tests helpers aren't very consistent, so take this with a grain of salt...
Generally when I have an 'ExpectingError' method I will wait for the error banner, or some other appropriate indicator, to show up and then I return either the error message or a reference to the error element. If no error was generated I return an empty string or a null.

Reply:
I've just removed these and a few other unused methods. There are still some unused methods left here, but they have better chance of working as written.